src/v/cluster/scheduling/leader_balancer_greedy.h

```diff
@@ -180,8 +193,10 @@ class greedy_balanced_shards final : private leader_balancer_strategy {
     void rebuild_load_index() {
         _load.clear();
         _load.reserve(_cores.size());
+        _load_map.reserve(_cores.size());
         for (auto it = _cores.cbegin(); it != _cores.cend(); ++it) {
             _load.push_back(it);
+            _load_map.emplace(it->first, it->second.size());
```

Comment:
We need this _load_map which maps from shard to the load on that shard because the indexing structure doesn't contain it. Rather than messing with the index definition and with the caller which has to build the index (who would then need to know what "load" means) we keep the same encapsulation as before and just build this load map as we are building the load array.